src/panels/lovelace/cards/hui-weather-forecast-card.ts

```diff
@@ -153,140 +127,127 @@ class HuiWeatherForecastCard extends LitElement implements LovelaceCard {
       `;
     }
 
-    const forecast = stateObj.attributes.forecast
-      ? stateObj.attributes.forecast.slice(0, 5)
+    if (stateObj.state === UNAVAILABLE) {
+      return html`
+        <ha-card class="unavailable" @click=${this._handleAction}>
+          ${this.hass.localize(
+            "ui.panel.lovelace.warning.entity_unavailable",
+            "entity",
+            `${computeStateName(stateObj)} (${this._config.entity})`
+          )}
+        </ha-card>
+      `;
+    }
+
+    const forecast = stateObj.attributes.forecast?.length
+      ? stateObj.attributes.forecast.slice(0, this._narrow ? 3 : 5)
       : undefined;
 
+    let hourly: boolean | undefined;
+
+    if (forecast?.length && forecast?.length > 1) {
+      const date1 = new Date(forecast[0].datetime);
+      const date2 = new Date(forecast[1].datetime);
+      const timeDiff = date2.getTime() - date1.getTime();
+
+      hourly = timeDiff < DAY_IN_MILLISECONDS;
+    }
+
     return html`
       <ha-card
         @action=${this._handleAction}
         .actionHandler=${actionHandler()}
         tabindex="0"
       >
-        <div class="header">
-          ${this.hass.localize(`state.weather.${stateObj.state}`) ||
-            stateObj.state}
-          <div class="name">
-            ${(this._config && this._config.name) || computeStateName(stateObj)}
-          </div>
-        </div>
         <div class="content">
-          <div class="now">
-            <div class="main">
-              ${stateObj.state in weatherIcons
-                ? html`
-                    <ha-icon icon="${weatherIcons[stateObj.state]}"></ha-icon>
-                  `
-                : ""}
-              <div class="temp">
-                ${stateObj.attributes.temperature}<span
-                  >${this.getUnit("temperature")}</span
-                >
+          <div class="icon-info">
+            ${stateObj.state in weatherIcons || stateObj.state in weatherImages
+              ? html`
+                  <state-badge
```

Comment:
Let's not use a `state-badge` if we are going to override everything, just use an `img` or `ha-icon`